### PR TITLE
Mark sacrebleu as supporting mypy type hints (see PEP 561)

### PR DESCRIPTION
**Pull request by martinpopel**
This should allow anyone to run `mypy my-code-using-sacrebleu.py`
and `mypy` will check all types.
Without the `py.typed` marker, `mypy` prints just:

```
error: Skipping analyzing 'sacrebleu': found module but no type hints or library stubs
note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
```

### CODE REVIEW

**Comment by ozancaglayan** [comment]
Do you want me to push a couple of mypy/annotation fixes to the code in this branch before merging?

**Comment by martinpopel** [comment]
@ozancaglayan: Yes, please add mypy/annotation fixes. I have not used mypy much so far and I won't have time for it soon.
It would be nice to add mypy tests to Travis, after fixing the errors (#100).

**Comment by mjpost** [comment]
Can you merge in `master`? Feel free to merge this, unless it is awaiting further changes from @ozancaglayan.

**Comment by ozancaglayan** [comment]
Let's re-run mypy and stash new fixes here after TER merge.

**Comment by martinpopel** [comment]
> Feel free to merge this

@mjpost GitHub does not allow me to merge this because "At least 1 approving review is required by reviewers with write access". Obviously, I cannot review my own PR.

@ozancaglayan: as I wrote, feel free to add mypy/annotation fixes here. I don't have time for it in the near future.

**Comment by ozancaglayan** [comment]
Working on this, a couple of fixes requires intervention by @ales-t for TER code. I'll paste them here.

**Comment by ozancaglayan** [comment]
Outstanding issues for TER:
```
sacrebleu/metrics/ter.py:415: error: Need type comment for '_cache' (hint: "_cache = ...  # type: Dict[<type>, <type>]")
sacrebleu/metrics/ter.py:464: error: Unsupported operand types for + ("List[List[Tuple[Any, ...]]]" and "List[List[Tuple[float, str]]]")
sacrebleu/metrics/ter.py:521: error: No overload variant of "__add__" of "tuple" matches argument type "str"
sacrebleu/metrics/ter.py:521: note: Possible overload variants:
sacrebleu/metrics/ter.py:521: note:     def __add__(self, Tuple[Any, ...]) -> Tuple[Any, ...]
```

File: sacrebleu/metrics/ter.py
Comment (ozancaglayan) [inline sacrebleu/metrics/ter.py:134]: Does this sound OK @ales-t ? `mypy` does not like a None variable being assigned something else dynamically.

Comment (ales-t) [reply sacrebleu/metrics/ter.py:134]: Looks good!

**Comment by ales-t** [comment]
Hi @ozancaglayan , regarding TER type hints, I made the necessary changes here:

https://github.com/martinpopel/sacreBLEU/compare/add-py-typed...ales-t:mypy-hints?expand=1

I'm not sure what the right approach is for contributing to someone else's PR but hopefully you can get the code from there easily.

**Comment by martinpopel** [comment]
I merged @ales-t's two commits into this PR. Once approved, it can be squash-merged.

**Comment by ales-t** [comment]
Sorry @martinpopel , I was too eager with the fixes. Please update your branch, I pushed a fix of the bug that I introduced. I ran tests locally and everything seems to be okay now.

**Review by ozancaglayan — APPROVED** [review]

**Comment by mjpost** [comment]
Sure, let's do that. Please be sure to simplify the commit message when you do the merge, so that it summarizes and is easy to read, but is not too detailed.

**Pull request merged**
